build_system/src/test.rs

```diff
@@ -982,7 +982,8 @@ where
         for path in files
             .iter()
             .enumerate()
-            .filter(|(pos, _)| *pos < start || *pos >= end)
+            .take(start)
+            .skip(count)
             .map(|(_, path)| path)
```

Comment:
```rust
files.iter().take(start).skip(count)
```

you don't need enumerate and map anymore.

Reply:
I asked in the issue, but let repeat it here: I do not think this can be done with skip and take, because the filter if I understand it correctly yields items from beginning of the files, skips some in the middle and then possibly yields additional items from end. And I do not know how to do this with take and skip.

Reply:
I believe the original code might be wrong. You probably want `.skip(start).take(count)`, though.

The code is supposed to take contiguous elements: that's why I think it might be wrong. Please use `.skip(start).take(count)` and I'll check if this is OK.

Reply:
I updated the code.

Reply:
The original code may be likely to be corresponding to 
```Rust
for path in files.iter().take(start).chain(files.iter().skip(end)) {
```
Which is weird.